src/algorithms/grassmann.jl

```diff
@@ -71,17 +71,19 @@ end
 function ManifoldPoint(state::Union{InfiniteMPS,FiniteMPS}, envs)
     al_d = similar(state.AL)
     @static if Defaults.parallelize_sites
+        g = fill(nothing, length(state))  #not typstable but this won't be a performance issue for now :) 
```

Comment:
Does this even work?
Also, can you not preallocate with a `g = similar(state.AL)`? I think these things should have the same type?

Reply:
come to think about it, do we even need this al_d separately? I think you can avoid allocating that vector and only keep the projected gradient